doc.md

```diff
@@ -0,0 +1,110 @@
+```
+FILES_LOCATION=/orca_repo/etc
+PROJECT=hackseq17_6
+
+docker run -t --name $PROJECT -m 100GB --memory-reservation 20GB -v $FILES_LOCATION/hosts:/etc/hosts:ro -v /projects/hackathon/2017:/home/projects -v $FILES_LOCATION/passwd_hs17:/etc/passwd:ro -v $FILES_LOCATION/group_hs17:/etc/group:ro -w /home/projects/$PROJECT -v /home/userhome:/home/userhome -v /tmp/.X11-unix:/tmp/.X11-unix --net host bcgsc/orca-hackseq:2017.10.12
+```
+
+*Details*:
+
+`--name $PROJECT`: a name of the container = a name of the team 
+
+`-m 100GB`: The maximum amount of memory that a container can use.
+`--memory-reservation 20GB`: a soft limit, which is activated when Docker detects low memory or contention on the host machine. A server has 128 GB RAM / 5 teams = ~20 GB.
+
+`-v $FILES_LOCATION/hosts:/etc/hosts:ro`: a flag `–h` doesn’t work with a `--net host`, so, have to mount hosts file.
+
+`-v /projects/hackathon/2017:/home/projects`: mounts all project directories 
+
+`-v $FILES_LOCATION/passwd_hs17:/etc/passwd:ro -v $FILES_LOCATION/group_hs17:/etc/group:ro`: maps users of the container with users of the host machine
+
+`-w /home/projects/$PROJECT`:a work directory will be a project folder
+
+`-v /home/userhome:/home/userhome`: mounts home directories of all users
+
+`/tmp/.X11-unix:/tmp/.X11-unix --net host`: for running GUI application (X11)
+
+## Log in
+When a user logs in to the system the following script (`/usr/local/bin/orca.hackathon`) runs:
+
+```
+#!/bin/sh
+	 
+	if [ "$1" = "-c" ];then
+	        cmd=`echo $2 | grep "rsync\|scp"`
+	        if [ -z "$cmd" ]; then
+	                echo "======================================================="
+	                echo "           Welcome to BCGSC ORCA Service          "
+	                echo "Only direct rsync or scp file transfer commands allowed"
+	                echo "======================================================="
+	        else
+	            	exec $2
+	        fi
+	else
+	 
+	        USER=$(id -un)
+	        TEAM=$(id -gn)
+	 
+	        echo "Welcome to Hackseq 2017, USER $(id -un), Your home directory is mounted under /home/userhome/$USER"
+	        echo "Currently installed tools can be found via 'brew list' and '/home/linuxbrew/.linuxbrew/bin' in shell"
+	        echo "You can remotely transfer data via rsync/scp to your /home/userhome/$USER directory"
+	        echo "Your Hackathon project folder is mounted under /home/projects/$TEAM, please keep your raw data, scripts here"
+	   
+	        exec docker exec -it  -u $(id -u):$(id -g) $(id -gn) /usr/bin/env XAUTHORITY=/home/userhome/$USER/.Xauthority DISPLAY=${DISPLAY} /bin/bash -l
+
+```
+
+*Details*:
+* The name of the container = the name of the primary group = `$(id -gn)`
+
+* `XAUTHORITY=/home/userhome/$USER/.Xauthority`: provides a path to user’s `.Xauthority` file, otherwise it takes root’ `.Xauthority` file, since the container was created by root.
```

Comment:
`since the container was created by root`. I think it would be better to create an `orca-admin` user on the host machine, and use that user to administrate, like creating containers. It's better to avoid using the `root` user whenever possible.

Reply:
Okay